src/VirtoCommerce.CatalogExportImportModule.Data/Validation/ImportPhysicalProductSkuValidator.cs

```diff
@@ -0,0 +1,70 @@
+{
+    public class ImportPhysicalProductSkuValidator : AbstractValidator<ImportRecord<CsvPhysicalProduct>>
+    {
+        private static readonly char[] ProductSkuIllegalCharacters = { '$', '+', ';', '=', '%', '{', '}', '[', ']', '|', '@', '~', '!', '^', '*', '&', '(', ')', '<', '>' };
+
+        public ImportPhysicalProductSkuValidator()
+        {
+            AttachValidators();
+        }
+
+        private void AttachValidators()
+        {
+            RuleFor(record => record.Record.ProductSku)
+                .NotEmpty()
+                .WithMissingRequiredValueCodeAndMessage("Product Sku")
+                .WithImportState()
+                .DependentRules(() =>
+                {
+                    RuleFor(record => record.Record.ProductSku)
+                        .MaximumLength(64)
+                        .WithExceededMaxLengthCodeAndMessage("Product Sku", 64)
+                        .WithImportState()
+                        .Must(sku => sku.IndexOfAny(ProductSkuIllegalCharacters) < 0)
+                        .WithInvalidValueCodeAndMessage("Product Sku")
+                        .WithImportState()
+                        .DependentRules(() =>
+                        {
+                            RuleFor(record => record.Record.ProductSku)
+                                .Must((record, sku, context) =>
+                                {
+                                    var existedProductsWithSameSku =
+                                        (CatalogProduct[])context.ParentContext.RootContextData[
+                                            ModuleConstants.ValidationContextData.ExistedProductsWithSameSku];
+
+                                    var productWithSuchSku = existedProductsWithSameSku.FirstOrDefault(x => x.Code.EqualsInvariant(sku));
+
+                                    if (productWithSuchSku == null)
+                                    {
+                                        return true;
+                                    }
+
+                                    if ( // updating of product case.
+                                         // do not check by outer id because id was set before validation if outer id exists
+                                        (!string.IsNullOrEmpty(record.Record.ProductId) &&
+                                         !productWithSuchSku.Id.EqualsInvariant(record.Record.ProductId))
+                                        // creating of product case
+                                        || (string.IsNullOrEmpty(record.Record.ProductId))
+                                    )
+                                    {
+                                        return false;
+                                    }
+
+                                    return true;
+                                })
+                                .WithMessage("Product with the same SKU and with different Id already exists in the current catalog.")
```

Comment:
Why don't you move the error message into the class with constants for more consistency?

Reply:
I delivery to QA so. While it testing i'll move messages to consts.